test/e2e-go/features/incentives/mining_test.go

```diff
@@ -0,0 +1,319 @@
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package suspension
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated/model"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/libgoal"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/test/framework/fixtures"
+	"github.com/algorand/go-algorand/test/partitiontest"
+)
+
+// first bonus payout, set in config/consensus.go
+const bonus1 = 10_000_000
+
+// TestBasicPayouts shows proposers getting paid
+func TestBasicPayouts(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	t.Parallel()
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	var fixture fixtures.RestClientFixture
+	// Make the seed lookback shorter, otherwise we need to wait 320 rounds to become IncentiveEligible.
+	fixture.FasterConsensus(protocol.ConsensusFuture)
+	fixture.Setup(t, filepath.Join("nettemplates", "Mining.json"))
+	defer fixture.Shutdown()
+
+	// Overview of this test:
+	// rereg to become eligible (must pay extra fee)
+	// show payouts are paid (from fees and bonuses)
+	// deplete feesink to ensure it's graceful
+
+	clientAndAccount := func(name string) (libgoal.Client, model.Account) {
+		c := fixture.GetLibGoalClientForNamedNode(name)
+		accounts, err := fixture.GetNodeWalletsSortedByBalance(c)
+		a.NoError(err)
+		a.Len(accounts, 1)
+		fmt.Printf("Client %s is %v\n", name, accounts[0].Address)
+		return c, accounts[0]
+	}
+
+	c15, account15 := clientAndAccount("Node15")
+	c01, account01 := clientAndAccount("Node01")
+	relay, _ := clientAndAccount("Relay")
+
+	data01 := rekeyreg(&fixture, a, c01, account01.Address)
+	data15 := rekeyreg(&fixture, a, c15, account15.Address)
+
+	// have account01 burn some money to get below the eligibility cap
+	// Starts with 100M, so burn 60M and get under 70M cap.
+	txn, err := c01.SendPaymentFromUnencryptedWallet(account01.Address, basics.Address{}.String(),
+		1000, 60_000_000_000_000, nil)
+	a.NoError(err)
+	burn, err := fixture.WaitForConfirmedTxn(uint64(txn.LastValid), txn.ID().String())
+	a.NoError(err)
+	data01, err = c01.AccountData(account01.Address)
+	a.NoError(err)
+
+	// Go 31 rounds after the burn happened. During this time, incentive
```

Comment:
is 31 = Payouts.MaxMarkAbsent (32) - 1? If so maybe to use `MaxMarkAbsent` and set it lower to speed up the test?

Reply:
No, it's `4*proto.SeedRefreshInterval-1`, that is, it's what would normally be 320-1, but `FasterConsensus` makes `SeeRefreshInterval`=8.

I'll pull that out of the consensus params so it's more self documenting.

I don't think I want to turn `SeeRefreshInterval` down even further, but I suppose I could.  What do you think?

Reply:
Pushed changed are good enough for me, thanks.